### PR TITLE
Added hca_bionetwork module to Project. Fixes #1517 

### LINKED ISSUE #1517: Update project tab with four metadata entities for grouping Tissue Atlas projects
### For which schema is a change/update being suggested?
Create a new 'hca_bionetwork' module and add it to project.json

### What should the change/update be?
Creating four fields as part of the 'hca_bionetwork' module and adding them to project.json

#### 1. HCA Bionetworks
I would like to add a new field - `Project.hca_bionetwork.name` - to this schema. This will allow data contributors to denote if the project is part of a HCA bionetwork and if so, which one.

This update constitutes a minor change to the schema(s) it affects.

```
Field name: project.hca_bionetwork.name
Field description: A field describing if the project is part of an HCA Bionetwork (e.g. Kidney). Enter ‘No’ if the project is not part of an HCA Bionetwork
Field type: array
Required: no
Examples: Kidney; Lung; No
CV or enum: enum
```

#### 2. HCA Tissue Atlas
I would like to add a new field - `Project.hca_bionetwork.hca_tissue_atlas` - to this schema. This will allow data contributors to denote if the project is part of a HCA Tissue Atlas and if so, which one. In this case, a Tissue Atlas represents a singular h5ad file. 

This update constitutes a minor change to the schema(s) it affects.

```
Field name: project.hca_bionetwork.hca_tissue_atlas
Field description: A field describing if the project is part of a HCA Tissue Atlas (e.g. Kidney). 
Field type: string, array
Required: no
Examples: Brain Alzheimer's Atlas 
CV or enum: Free Text
```

#### 3. HCA Tissue Atlas Version
I would like to add a new field - `Project.hca_bionetwork.hca_tissue_atlas_version` - to this schema. This will allow data contributors to denote what version of the Tissue Atlas is associated with this publication (e.g. v1.0; v2.0)

This update constitutes a minor change to the schema(s) it affects.

```
Field name: project.hca_bionetwork.hca_tissue_atlas_version
Field description: A field describing the version of the HCA Tissue Atlas is associated with the publication (e.g. v1.0; v2.0)
Field type: string, array
Required: no
Examples: 'v1.0; v2.0'
CV or enum: REGEX
```

#### 2. Integrated project
I would like to add a new field - `Project.hca_bionetwork.atlas_project` - to this schema. This will allow data contributors to denote if the publication is the HCA Tissue Atlas publication which integrates data from other projects within the atlas.

This update constitutes a minor change to the schema(s) it affects.

```
Field name: project.hca_bionetwork.atlas_project
Field description: A field describing if the project integrates data from other publications. 
Enter ‘Yes’ if this project is a HCA Tissue Atlas and it integrates data from all other datasets. Enter ‘No’, if this project's data is being integrated.
Field type: string
Required: no
Examples: 'Yes; No'
CV or enum: enum
```

### Motivation
Biological Network Data Contributors need to be able to denote if 
a) Their project is part of a HCA Bionetwork
b) Their project is part of a HCA Tissue Atlas
c) What version of the HCA Tissue Atlas the project is a part of
b) Whether the project is the 'Atlas Project' which draws data from other projects, or the project is a project from which data is being integrated

### PR DESCRIPTION
**Pull request by Wkt8**
### Release notes
Added module/project/hca_bionetwork.json schema:
- Added `names` field
- Added `hca_tissue_atlas` field
- Added `hca_tissue_atlas_version` field
- added `atlas_project` field

### Why are these changes needed?

This change is needed for data contributors and consumers, to describe if this project is part of a HCA bionetwork, is part of a HCA Atlas, what version of the atlas, and whether it is the 'Atlas Project' containing a h5ad file that draws from other projects or not. 

### Reviews requested

- Need Reviewers to approve because this is a minor update

Discussion here: https://miro.com/app/board/uXjVMfMlgLc=/
We have planned the four metadata schema fields to explicitly describe the metadata of the project. [Wrangler meeting notes](https://docs.google.com/document/d/1N3dJA5dQuJcY5pxhT2pNgwZptKfdH3BSsqZSnWD0nUo/edit#)
[Example spreadsheet](https://docs.google.com/spreadsheets/d/1CiyCJZ96lu3PBLr7RAgbz08h1qA66n0Z/edit?usp=sharing&ouid=114069941208994528669&rtpof=true&sd=true)

### CODE REVIEW

File: json_schema/module/project/hca_bionetwork.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            "type": "string",
+            "pattern": "v[0-9].[0-9]",
+            "user_friendly": "Official HCA Tissue Atlas Version",
+            "guidelines": "For example: v1.0; v2.0",
+            "example": "v1.0; v2.0"
+        },
+        "atlas_project": {
+            "description": "A field describing if this project is the HCA Tissue Atlas project which integrates data from other datasets.",
+            "type": "string",
+            "enum": ["Yes","No"],
```
Comment (hannes-ucsc) [inline json_schema/module/project/hca_bionetwork.json:69]:
```suggestion
            "type": "boolean",
```

right?

File: json_schema/module/project/hca_bionetwork.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    "properties": {
+        "describedBy": {
+            "description": "The URL reference to the schema.",
+            "type": "string",
+            "pattern" : "^(http|https)://schema.(.*?)humancellatlas.org/module/project/(([0-9]{1,}.[0-9]{1,}.[0-9]{1,})|([a-zA-Z]*?))/contact"
+        },
+        "schema_version": {
+            "description": "The version number of the schema in major.minor.patch format.",
+            "type": "string",
+            "pattern": "^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$",
```
Comment (hannes-ucsc) [inline json_schema/module/project/hca_bionetwork.json:19]:
```suggestion
            "pattern": "^[0-9]+\\.[0-9]+\\.[0-9]+$",
```

File: json_schema/module/project/hca_bionetwork.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            "description": "A field describing if the project is part of a HCA Tissue Atlas (e.g. Brain Alzheimer Atlas). ",
+            "type": "string",
+            "user_friendly": "HCA Tissue Atlas",
+            "guidelines": "For example: Brain Alzheimer Atlas",
+            "example": "Brain Alzheimer Atlas"
+	},
+        "hca_tissue_atlas_version": {
+            "description": "A field describing which version of the HCA Tissue Atlas is associated with the projet (e.g. v1.0; v2.0)",
+            "type": "string",
+            "pattern": "v[0-9].[0-9]",
```
Comment (hannes-ucsc) [inline json_schema/module/project/hca_bionetwork.json:61]:
```suggestion
            "pattern": "^v[0-9]+\\.[0-9]+$",
```

But even with these corrections, the pattern would allow non-sensical strings like v0000.09

I recommend breaking this up into two separate integer properties, one for the major version and one for the minor version.

File: json_schema/module/project/hca_bionetwork.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                "Liver",
+                "Lung",
+                "Musculoskeletal",
+                "Nervous System",
+                "Oral & Craniofacial",
+                "Organoid",
+                "Pancreas",
+                "Reproduction",
+                "Skin",
+                "No"
```
Comment (hannes-ucsc) [inline json_schema/module/project/hca_bionetwork.json:45]:
```suggestion
                "None"
```

 or 
 
 ```suggestion
                "N/A"
```

File: json_schema/module/project/hca_bionetwork.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            ],
+            "user_friendly": "Official HCA Bionetwork(s)",
+            "guidelines": "Should be one of the networks from https://www.humancellatlas.org/biological-networks/ or 'No' if not a network",
+            "example": "Kidney; Lung; No"
+        },
+        "hca_tissue_atlas": {
+            "description": "A field describing if the project is part of a HCA Tissue Atlas (e.g. Brain Alzheimer Atlas). ",
+            "type": "string",
```
Comment (hannes-ucsc) [inline json_schema/module/project/hca_bionetwork.json:53]: Fields like this are bound to end up with misspelled, inconsistently spelled values. Can you make this an enum or do you have something else planned to quality-control this field?

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: json_schema/module/project/hca_bionetwork.json [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            "pattern" : "^(http|https)://schema.(.*?)humancellatlas.org/module/project/(([0-9]{1,}.[0-9]{1,}.[0-9]{1,})|([a-zA-Z]*?))/contact"
+        },
+        "schema_version": {
+            "description": "The version number of the schema in major.minor.patch format.",
+            "type": "string",
+            "pattern": "^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$",
+            "example": "4.6.1"
+        },
+        "Names": {
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:22]:
```suggestion
        "name": {
```

File: json_schema/module/project/hca_bionetwork.json [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                "Musculoskeletal",
+                "Nervous System",
+                "Oral & Craniofacial",
+                "Organoid",
+                "Pancreas",
+                "Reproduction",
+                "Skin",
+                "No"
+            ],
+            "user_friendly": "Official HCA Bionetwork(s)",
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:47]:
```suggestion
            "user_friendly": "Official HCA Bionetwork",
```

File: json_schema/module/project/hca_bionetwork.json (hunk 5 above)
Comment (ESapenaVentura) [reply json_schema/module/project/hca_bionetwork.json:53]: @hannes-ucsc I think we are currently discussing about this. So far (Please @Wkt8 and @idazucchi correct me if I'm wrong) we have no official listing.

File: json_schema/module/project/hca_bionetwork.json (hunk 3 above)
Comment (ESapenaVentura) [reply json_schema/module/project/hca_bionetwork.json:61]: would 
```suggestion
            "pattern": "^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$",
```

work better? Although I am not opposed against creating a "major" and "minor" version, we should probably discuss about this @Wkt8

File: json_schema/update_log.csv [hunk 8]
```diff
@@ -1 +1,3 @@
 Schema,Change type,Change message,Version,Date
+module/project/hca_bionetwork,minor,"Added the hca_bionetwork module with four fields for recording bionetwork and atlas information",,
```
Comment (ESapenaVentura) [inline json_schema/update_log.csv:2]: https://github.com/HumanCellAtlas/metadata-schema/blob/master/docs/committers.md#specific-how-to-for-making-changes , point 5, new schemas only

tl;dr, this should be marked as a major change because we're adding a module. For the project entity this is minor, but the module should start in version 1.0.0 :)

**Review by ESapenaVentura — CHANGES_REQUESTED** [review]
Left some comments!

File: json_schema/type/project/project.json [hunk 9]
```diff
@@ -66,6 +66,14 @@
...
             "user_friendly": "Publications"
         },
+        "hca_bionetwork": {
+            "description": "HCA Bionetworks and Atlases the project is associated with",
+            "type": "array",
+            "items": {
+                "$ref": "module/project/hca_bionetwork.json"
+            },
+            "user_friendly": "HCA Bionetwork"
+        },
```
Comment (ESapenaVentura) [inline json_schema/type/project/project.json:76]:
```suggestion
        "hca_bionetworks": {
            "description": "HCA Bionetworks and Atlases the project is associated with",
            "type": "array",
            "items": {
                "$ref": "module/project/hca_bionetwork.json"
            },
            "user_friendly": "HCA Bionetwork(s)"
        },
```

File: json_schema/module/project/hca_bionetwork.json [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            "pattern" : "^(http|https)://schema.(.*?)humancellatlas.org/module/project/(([0-9]{1,}.[0-9]{1,}.[0-9]{1,})|([a-zA-Z]*?))/contact"
+        },
+        "schema_version": {
+            "description": "The version number of the schema in major.minor.patch format.",
+            "type": "string",
+            "pattern": "^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$",
+            "example": "4.6.1"
+        },
+        "Names": {
+            "description": "A field describing which HCA Bionetwork the project is a part of (e.g. Kidney). Enter ‘No’ if the project is not part of an official HCA Bionetwork.",
```
Comment (Wkt8) [inline json_schema/module/project/hca_bionetwork.json:23]:
```suggestion
            "description": "A field describing which HCA Bionetwork the project is a part of (e.g. Kidney). Enter ‘None’ if the project is not part of an official HCA Bionetwork.",
```

File: json_schema/module/project/hca_bionetwork.json (hunk 5 above)
Comment (idazucchi) [reply json_schema/module/project/hca_bionetwork.json:53]: We are hoping to get a list of the first few tissue atlases after the meeting on Friday with the advisory board, but this enum field will have to be updated in the future as the atlases are assembled.

File: json_schema/module/project/hca_bionetwork.json (hunk 3 above)
Comment (Wkt8) [reply json_schema/module/project/hca_bionetwork.json:61]: I think also because the bionetworks haven't given us any guidelines on 'what a major or minor version looks like' we're trying to future-proof by adding in major and minor version storage in our metadata schema, but we don't even know if they'll have major and minor versions.

File: json_schema/update_log.csv (hunk 8 above)
Comment (Wkt8) [inline json_schema/update_log.csv:2]:
```suggestion
module/project/hca_bionetwork,major,"Added the hca_bionetwork module with four fields for recording bionetwork and atlas information",,

```

File: json_schema/module/project/hca_bionetwork.json (hunk 5 above)
Comment (hannes-ucsc) [reply json_schema/module/project/hca_bionetwork.json:53]: If we could settle on an initial list of possible values, I'd obviously prefer an enum field. PRs that add entries to enums will be easy to review and merge.

**Review by ncalvanese1 — APPROVED** [review]

File: json_schema/module/project/hca_bionetwork.json [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        },
+        "hca_tissue_atlas": {
+            "description": "A field describing if the project is part of a HCA Tissue Atlas (e.g. Brain Alzheimer Atlas). ",
+            "type": "string",
+            "user_friendly": "HCA Tissue Atlas",
+            "guidelines": "For example: Brain Alzheimer Atlas",
+            "example": "Brain Alzheimer Atlas"
+	},
+        "hca_tissue_atlas_version": {
+            "description": "A field describing which version of the HCA Tissue Atlas is associated with the projet (e.g. v1.0; v2.0)",
```
Comment (Wkt8) [inline json_schema/module/project/hca_bionetwork.json:59]:
```suggestion
            "description": "A field describing which version of the HCA Tissue Atlas is associated with the project (e.g. v1.0; v2.0)",
```

File: json_schema/module/project/hca_bionetwork.json (hunk 5 above)
Comment (Wkt8) [reply json_schema/module/project/hca_bionetwork.json:53]: This field has been changed to an enum with three initial values - Blood, Retina, and None. More to come as we learn more from the Bionetworks!

File: json_schema/module/project/hca_bionetwork.json [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+            "pattern" : "^(http|https)://schema.(.*?)humancellatlas.org/module/project/(([0-9]{1,}.[0-9]{1,}.[0-9]{1,})|([a-zA-Z]*?))/contact"
+        },
+        "schema_version": {
+            "description": "The version number of the schema in major.minor.patch format.",
+            "type": "string",
+            "pattern": "^[0-9]+\\.[0-9]+\\.[0-9]+$",
+            "example": "4.6.1"
+        },
+        "name": {
+            "description": "A field describing which HCA Bionetwork the project is a part of (e.g. Kidney). Enter ‘None’ if the project is not part of an official HCA Bionetwork.",
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:23]:
```suggestion
            "description": "Name of the HCA Bionetwork the associated project is a part of (e.g. Kidney).",
```

I wouldn't use "A field ..." for the description, there's nothing in the guidelines but it sounds a little weird to me! Also the `Enter ‘None’ if the project is not part of an official HCA Bionetwork.` seems more fit in the guidelines :)

File: json_schema/module/project/hca_bionetwork.json [hunk 13]
```diff
@@ -0,0 +1,79 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "description": "Information about whether the project is part of a HCA Bionetwork or HCA Atlas.",
+    "additionalProperties": false,
+    "required": [
+    ],
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:6]: Won't there be any required property? If so, we can just delete this field

File: json_schema/module/project/hca_bionetwork.json [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+                "Liver",
+                "Lung",
+                "Musculoskeletal",
+                "Nervous System",
+                "Oral & Craniofacial",
+                "Organoid",
+                "Pancreas",
+                "Reproduction",
+                "Skin",
+                "None"
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:45]:
```suggestion
                "Skin"
```
If this field is not required, I would delete "None" from the list!

File: json_schema/module/project/hca_bionetwork.json [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+            ],
+            "user_friendly": "Official HCA Bionetwork",
+            "guidelines": "Should be one of the networks from https://www.humancellatlas.org/biological-networks/ or 'No' if not a network",
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:48]:
```suggestion
            "guidelines": "Should be one of the networks from https://www.humancellatlas.org/biological-networks/ or 'None' if not a network",
```

Also we can delete this depending on the comment above

File: json_schema/module/project/hca_bionetwork.json [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+            "example": "Kidney; Lung; No"
+        },
+        "hca_tissue_atlas": {
+            "description": "A field describing if the project is part of a HCA Tissue Atlas (e.g. Brain Alzheimer Atlas). ",
+            "type": "string",
+            "enum": [
+                "Blood",
+                "Retina",
+                "None"
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:57]: If this field is not required, I would not add the "None" option

File: json_schema/module/project/hca_bionetwork.json [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+            "user_friendly": "Official HCA Tissue Atlas Version",
+            "guidelines": "For example: v1.0; v2.0",
+            "example": "v1.0; v2.0"
+        },
+        "atlas_project": {
+            "description": "A field describing if this project is the HCA Tissue Atlas project which integrates data from other datasets.",
+            "type": "boolean",
+            "user_friendly": "Project Tissue Atlas Status",
+            "guidelines": "Enter ‘Yes’ if this project is one of the HCA Tissue Atlases and it integrates data from all other datasets. Enter ‘No’, if this project's data is being integrated.",
+            "example": "Atlas Project; No"
```
Comment (ESapenaVentura) [inline json_schema/module/project/hca_bionetwork.json:76]:
```suggestion
            "example": "Yes; No"
```

**Review by ESapenaVentura — CHANGES_REQUESTED** [review]
Added some comments!

Please take special attention to the field description comment, it applies to all of the descriptions for the newly created fields

File: json_schema/module/project/hca_bionetwork.json (hunk 14 above)
Comment (hannes-ucsc) [reply json_schema/module/project/hca_bionetwork.json:42]: > If this field is not required, I would delete "None" from the list!

(from https://github.com/HumanCellAtlas/metadata-schema/pull/1520/files#r1166781046)

What would be the difference, semantically speaking, between `"name": "None"` and the `name` property being absent? Downstream in the Data Browser this would render as `Unspecified` vs `None`. If this distinction isn't meaningful, then I'd have to agree with @ESapenaVentura.

**Review by NoopDog — APPROVED** [review]

**Review by ESapenaVentura — APPROVED** [review]
LGTM

**Review by hannes-ucsc — APPROVED** [review]
